llarp/util/threading.cpp

```diff
@@ -16,14 +19,18 @@ namespace llarp
 #ifdef POSIX
 #ifdef __MACH__
       const int rc = pthread_setname_np(name.c_str());
+#else
+#ifdef __FreeBSD__
+      pthread_setname_np(pthread_self(), name.c_str());
```

Comment:
based on https://www.freebsd.org/cgi/man.cgi?query=pthread_setname_np&sektion=3&apropos=0&manpath=OSF1+V5.1%2falpha

this should be:

```c++
const int rc = pthread_setname_np(pthread_self(), name.c_str(), nullptr);
```